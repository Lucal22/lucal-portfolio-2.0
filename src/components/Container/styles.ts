import styled, { css } from 'styled-components';

export type ContainerStyleProps = {
  background: 'white' | 'gray';
};

export const Container = styled.div<ContainerStyleProps>`
  ${({ theme, background }) => css`
    margin: 0 auto;
    min-height: 100vh;
    width: ${theme.screen.size.xSmall};
    background-color: ${background == 'white'
      ? theme.background.white
      : theme.background.gray};
    color: ${background == 'white'
      ? theme.colors.blackColor
      : theme.colors.whiteColor};
    @media (min-width: ${theme.screen.size.small}) {
      width: ${theme.screen.size.small};
    }
    @media (min-width: ${theme.screen.size.medium}) {
      width: ${theme.screen.size.medium};
    }
    @media (min-width: ${theme.screen.size.large}) {
      width: ${theme.screen.size.large};
    }
  `}
`;
