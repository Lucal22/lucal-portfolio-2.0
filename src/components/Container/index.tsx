import React from 'react';
import * as Styled from './styles';
import { ContainerStyleProps } from './styles';

export type ContainerProps = ContainerStyleProps & {
  children: React.ReactNode;
};

export default function Container({ children, background }: ContainerProps) {
  return (
    <Styled.Container background={background}>{children}</Styled.Container>
  );
}
