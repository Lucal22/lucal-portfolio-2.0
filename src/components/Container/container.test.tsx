import { screen } from '@testing-library/react';
import { renderTheme } from '../../utils/renderTheme';
import Container from './index';

describe('<Container />', () => {
  it('should render with background white', () => {
    const { container } = renderTheme(
      <Container background={'white'}>
        <h1>Olá mundo</h1>
      </Container>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument;
    expect(container).toMatchSnapshot();
  });

  it('should render with background gray', () => {
    const { container } = renderTheme(
      <Container background={'gray'}>
        <h1>Olá mundo</h1>
      </Container>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument;
    expect(container).toMatchSnapshot();
  });
});
