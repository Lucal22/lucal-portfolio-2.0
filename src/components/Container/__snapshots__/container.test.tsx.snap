// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render with background gray 1`] = `
.c0 {
  margin: 0 auto;
  min-height: 100vh;
  width: 360px;
  background-color: #6b7688;
  color: #ffffff;
}

@media (min-width:481px) {
  .c0 {
    width: 481px;
  }
}

@media (min-width:780px) {
  .c0 {
    width: 780px;
  }
}

@media (min-width:1280px) {
  .c0 {
    width: 1280px;
  }
}

<div>
  <div
    class="c0"
  >
    <h1>
      Olá mundo
    </h1>
  </div>
</div>
`;

exports[`<Container /> should render with background white 1`] = `
.c0 {
  margin: 0 auto;
  min-height: 100vh;
  width: 360px;
  background-color: #ffffff;
  color: #030303;
}

@media (min-width:481px) {
  .c0 {
    width: 481px;
  }
}

@media (min-width:780px) {
  .c0 {
    width: 780px;
  }
}

@media (min-width:1280px) {
  .c0 {
    width: 1280px;
  }
}

<div>
  <div
    class="c0"
  >
    <h1>
      Olá mundo
    </h1>
  </div>
</div>
`;
